src/components/MapsSection/MapsSection.tsx

```diff
@@ -102,6 +104,57 @@ const MapsSection = ({
     });
   }, [currentVisu]);
 
+  useEffect(() => {
+    const button = buttonRef.current;
+    const container = tagsContainerRef.current;
+
+    if (container && button) {
+      const currentIndex = tiffInfo.findIndex(
+        (map) => map.fields.id === currentVisu.id,
+      );
+
+      const containerTop = container.scrollTop;
+      const containerBottom = containerTop + container.clientHeight;
+
+      const buttonOffsetHeight = button.offsetHeight;
+      const buttonHeight = button.offsetTop;
+      +buttonOffsetHeight;
+
+      let tamBotao = buttonHeight / 100;
+      const isMobile = window.innerWidth <= 810;
+      const screen = window.innerHeight;
+      const maxVisibleBoxes = isMobile ? 1 : 5;
+      const needsScroll =
+        (buttonHeight / 9) * currentIndex > containerBottom ||
+        currentIndex === 0;
+
+      if (needsScroll) {
+        const scrollDesktop = Math.abs(
+          tamBotao + (currentIndex - maxVisibleBoxes) * buttonOffsetHeight,
+        );
+        const scrollToPositionDesktop = currentIndex === 0 ? 0 : scrollDesktop;
+
+        container.scrollTo({
+          top: scrollToPositionDesktop,
+          behavior: "smooth",
+        });
+      }
+
+      if (isMobile && needsScroll) {
+        tamBotao = buttonHeight / 10 + buttonOffsetHeight;
+        const scrollMobile = Math.abs(
+          (tamBotao * currentIndex) / (screen / 370),
+        );
```

Comment:
How did you came up with this solution? Just curious

Reply:
Honestly, this was more trial and error. Since I already had scrollDesktop working well, I tried adapting it for mobile using the same variables. I also noticed that the screen height was important for determining the scroll position, so I compared some values until i get screen / 370, which works well for all the tests I did

Reply:
By the way, I would love to hear your suggestions for improving this part of the code and the previous ones. I honestly believe I did some things that are unnecessary, but I didn’t know how to do better at the time